pkg/pf/tests/internal/cross-tests/tfwrite.go

```diff
@@ -0,0 +1,169 @@
+package crosstests
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	pschema "github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	rschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type PFWriter struct{ out io.Writer }
+
+func WritePF(out io.Writer) PFWriter { return PFWriter{out} }
+
```

Comment:
We don't need a `WritePF` abstraction if everything is already in PF land. These could just be `writeProvider`, `writeResource`, etc.

Also, these shouldn't be public.